File: testdata/bench/moving_avg.lua
local n = 200000
local window = 10

local data = {}
for i = 0, n - 1 do
    table.insert(data, (i * 13) % 1000)
end

local sum = 0
for i = 1, window do
    sum = sum + data[i]
end

local out = 0
for i = window + 1, n do
    sum = sum + data[i]
    sum = sum - data[i - window]
    out = out + math.floor(sum / window)
end

print(out)

File: testdata/bench/moving_avg.js
var n = 200000;
var w = 10;

var data = [];
for (var i = 0; i < n; i++) {
    data.push((i * 13) % 1000);
}

var sum = 0;
for (var i = 0; i < w; i++) {
    sum += data[i];
}

var out = 0;
for (var i = w; i < n; i++) {
    sum += data[i];
    sum -= data[i - w];
    out += Math.floor(sum / w);
}

console.log(out);
